### PR TITLE
Migrate to pyproject.toml and modernize a little bit the CI

### PR DESCRIPTION
**Pull request by Terseus**
This PR arises from a discussion at  #503.

Changes summary:

* Now the CI builds the package and install it from the package before running the tests; this should catch any errors introduced in the package imports or in the package build system.
* Drops Python 3.6 support; unfortunately I had no choice if we wanted to migrate to pyproject.toml because Python 3.6 doesn't support pyproject.toml, however given that Python 3.6 reached end of line at the beginning of 2022 (see [PEP 494](https://peps.python.org/pep-0494/#lifespan)) I don't think is a big deal.
* Add all the current Python versions to CI, up to the new and shiny 3.11.
* Upgrade CI python-setup action to v4.
* Introduce the `requirements-dev.txt` file with the requirements needed for development.
* Refactor the existing tests from class-based to function-based, mostly to prevent accidental coupling between them.
* Add `TooManyRequestsError` and mark all the tests as xfail to prevent the CI failing when Google decides we're not worthy.

@emlazzarin This is the first big step before starting to create a more robust unit test suite that doesn't call **always** the backend API in every test execution.
The plan was just migrate to pyproject.toml but I got a bit carried away :)

Comments are more than welcome.

**Edit**: Oh I forgot to add something important, there are some differences in the `METADATA` file generated:
* The "Home-Page" directive is not generated anymore, instead an URL with the name "homepage" is generated; in theory this is equivalent (see [this discussion at setuptools](https://github.com/pypa/setuptools/discussions/3318) and [this discussion at pypi warehouse](https://github.com/pypi/warehouse/issues/11220)) but I'm not 100% sure, maybe we should upload to [https://test.pypi.org](https://test.pypi.org/account/register/) before merging?
* The "Author" directive it's not an array of strings but two strings separated by comma, hope it's equivalent to an array; BTW should we assign the `dreyco676` to any on the two authors?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    {email = "dreyco676@gmail.com"}
+]
+license = {text = "Apache 2.0"}
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 3.3",
+    "Programming Language :: Python :: 3.4",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
```
Comment (emlazzarin) [inline pyproject.toml:23]: I'm okay with removing these.

**Review by emlazzarin — COMMENTED** [review]
I wonder if we should merge these into a new `dev` branch before going into `master`. We may have an opportunity to clean up a bunch of things at once and make a bigger release.

File: pyproject.toml (hunk 1 above)
Comment (Terseus) [reply pyproject.toml:23]: Oops, an oversight, I'll update them.

**Comment by Terseus** [comment]
Personally I prefer smaller releases but I'll adapt to your decisions.

However before doing release, please consider doing version tagging (starting with tagging the current v4.8.0) and doing releases; if you want we can start using a `CHANGELOG.md` file to ease writing the releases changelog, starting in this PR.

My next step is writing unit tests that cover the whole library functionality and doesn't call to the backend at every CI execution.
Once we have this we can start cleaning up the project with more confidence.

File: pyproject.toml (hunk 1 above)
Comment (Terseus) [reply pyproject.toml]: I fixed more things in the `pyproject.toml`:
* updated the license classifier to match with the `LICENSE` file.
* updated the version to match the latest release at [pypi](https://pypi.org/project/pytrends/) (4.8.0).

**Pull request merged**